From 10e3570ab00aac997c6e428106acfe3e0eface08 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Wed, 15 Nov 2017 17:11:40 +0800
Subject: [PATCH 2808/5242] MLK-16830 PCI: imx: fix boot failulre when the pll
 is not locked

commit  ef5e1176c5c47d968031ab3b2d4e1c643c0b23f0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Return the error when the pll is not locked.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Reviewed-by: Frank Li <frank.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index b22f33f..0d8d9ef 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -791,8 +791,10 @@ static int imx6_pcie_deassert_core_reset(struct imx6_pcie *imx6_pcie)
 			dev_err(dev, "ERROR PM_REQ_CORE_RST is still set.\n");
 
 		/* wait for phy pll lock firstly. */
-		if (imx8_pcie_wait_for_phy_pll_lock(imx6_pcie))
+		if (imx8_pcie_wait_for_phy_pll_lock(imx6_pcie)) {
 			ret = -ENODEV;
+			break;
+		}
 
 		/* set up the cpu address offset */
 		if (imx6_pcie->cpu_base)
@@ -832,7 +834,8 @@ static int imx6_pcie_deassert_core_reset(struct imx6_pcie *imx6_pcie)
 		mdelay(20);
 	}
 
-	return ret;
+	if (ret == 0)
+		return ret;
 
 err_ref_clk:
 	clk_disable_unprepare(imx6_pcie->pcie_phy);
-- 
1.7.9.5

